app/src/androidTest/java/com/epfl/drawyourpath/mainpage/fragments/ChallengeFragmentTest.kt

```diff
@@ -1,380 +1,324 @@
 package com.epfl.drawyourpath.mainpage.fragments
 
+import android.content.Context
 import android.os.Bundle
 import android.view.View
+import androidx.arch.core.executor.testing.CountingTaskExecutorRule
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
 import androidx.fragment.app.testing.FragmentScenario
 import androidx.recyclerview.widget.RecyclerView
-import androidx.test.espresso.Espresso.closeSoftKeyboard
+import androidx.test.core.app.ApplicationProvider
 import androidx.test.espresso.Espresso.onView
 import androidx.test.espresso.UiController
 import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.ViewActions.pressImeActionButton
 import androidx.test.espresso.action.ViewActions.replaceText
 import androidx.test.espresso.assertion.ViewAssertions.matches
-import androidx.test.espresso.contrib.RecyclerViewActions
+import androidx.test.espresso.contrib.RecyclerViewActions.actionOnItemAtPosition
+import androidx.test.espresso.contrib.RecyclerViewActions.scrollToPosition
 import androidx.test.espresso.matcher.ViewMatchers.*
 import androidx.test.ext.junit.runners.AndroidJUnit4
 import com.epfl.drawyourpath.R
-import com.epfl.drawyourpath.challenge.DailyGoal
 import com.epfl.drawyourpath.challenge.TemporaryUser
-import com.epfl.drawyourpath.community.Tournament
 import com.epfl.drawyourpath.challenge.Trophy
+import com.epfl.drawyourpath.community.Tournament
+import com.epfl.drawyourpath.database.MockDataBase
+import com.epfl.drawyourpath.userProfile.dailygoal.DailyGoal
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 import java.time.LocalDate
 import java.time.LocalDateTime
 import java.util.*
+import java.util.concurrent.TimeUnit
 
 
 @RunWith(AndroidJUnit4::class)
 class ChallengeFragmentTest {
 
-    private var userOnlyGoal =
-        TemporaryUser(LinkedList(mutableListOf(DailyGoal(5.0, 15.0, 1, 1.56, 12.0, 1))))
+    @get:Rule
+    val executorRule = CountingTaskExecutorRule()
 
+    private var mockUser = MockDataBase().userModelTest
+    private val context: Context = ApplicationProvider.getApplicationContext()
+    private val dailyGoal = DailyGoal.TEST_SAMPLE
 
     /**
-     * test if the distance goal is correctly displayed
+     * wait for all thread to be done or throw a timeout error
      */
-    @Test
-    fun displayCorrectDistanceGoal() {
-        val bundle = Bundle()
-        bundle.putSerializable("user", userOnlyGoal)
+    private fun waitUntilAllThreadAreDone() {
+        executorRule.drainTasks(2, TimeUnit.SECONDS)
+    }
 
+    /**
+     * test if the new distance goal is correctly displayed when there is no today daily goal
+     */
+    @Test
+    fun displayNewDistanceGoalWhenNoTodayDailyGoal() {
         val scenario = FragmentScenario.launchInContainer(
             ChallengeFragment::class.java,
-            bundle,
+            Bundle(),
             R.style.Theme_Bootcamp

```

Comment:
If the Bundle() is not use , you can call the function without it.

Reply:
I tested just now without it but it crashed because it had no theme and there is no constructor with a theme and no bundle